azext_edge/edge/providers/orchestration/connected_cluster.py

```diff
@@ -48,14 +105,38 @@ def extensions(self) -> List[dict]:
         return additional_properties.get("value", [])
 
     def get_custom_location_for_namespace(self, namespace: str) -> Optional[dict]:
-        query = f"""
-        resources
-        | where type =~ 'microsoft.extendedlocation/customlocations'
-        | where properties.hostResourceId =~ '{self.resource_id}'
-        | where properties.namespace =~ '{namespace}'
-        | project id, name, location, properties, apiVersion
-        """
+        query = QUERIES["get_custom_location_for_namespace"].format(resource_id=self.resource_id, namespace=namespace)
+
+        result = self.resource_graph.query_resources(query=query)
+        return self._process_query_result(result, first=True)
```

Comment:
can there be multiple custom locations? if yes, do we not care about getting all of them? if not, how do we determine the one to care about (assuming it is not always the first one)?

Reply:
There can only be 1 custom location associated to a namespace at a time. BTW `get_custom_location_for_namespace` is not used as part of the delete AIO operation, this was existing functionality support an `init`  pre-flight check, just refactoring here to conform with the updated pattern.